packages/webview_flutter/ios/Classes/FlutterWebView.m

```diff
@@ -289,6 +311,20 @@ - (bool)loadUrl:(NSString*)url withHeaders:(NSDictionary<NSString*, NSString*>*)
   return true;
 }
 
+- (bool)loadAssetFile:(NSString*)url {
+  NSString* key = [_registrar lookupKeyForAsset:url];
```

Comment:
It won't work if the url has search or hash parameters.

Reply:
> It won't work if the url has search or hash parameters.

Yes.. I try to load reactjs app from local file system.  it is not loading url with bookmark (#).  is there any work-around to load bookmark urls?

Reply:
yes.. try to load bookmar url.. it is not working. . is there any work-around?

Reply:
> yes.. try to load bookmark url.. it is not working. . is there any work-around?

The returned key of method `lookupKeyForAsset` only can be found when the `url` parameter is identical to the assets path configration in pubspec.yaml.
So before calling this method, we have to remove the search and hash parts in the `url` string.
But we can append them back to  `nsUrl`, the absolute path, before calling `loadFileURL`.

Reply:
thanks for your response..  i'm new to objective c.. tried the proposed changes you described above.. still getting same error..  My changes are in
https://github.com/kittugit/webview_flutter/blob/master/ios/Classes/FlutterWebView.m
if i use "fullKey", instead of key, i'm getting error. 

from console log: 
key: Frameworks/App.framework/flutter_assets/assets/build/index.html
fullKey: Frameworks/App.framework/flutter_assets/assets/build/index.html/#home

Reply:
```objc
- (bool)loadAssetFile:(NSString*)url {
  NSArray* array = [url componentsSeparatedByString:@"?"];
  NSString* pathString = [array objectAtIndex:0];
  NSLog(@"%@%@", @"pathString: ", pathString);
  NSString* key = [_registrar lookupKeyForAsset:pathString];
  NSURL* baseURL = [[NSBundle mainBundle] URLForResource:key withExtension:nil];
  if (!baseURL) {
      return false;
   }
  NSURL* newUrl = baseURL;
  if ([array count] > 1) {
    NSString* queryString = [array objectAtIndex:1];
    NSLog(@"%@%@", @"queryString: ", queryString);
    NSString* queryPart = [NSString stringWithFormat:@"%@%@", @"?", queryString];
    NSLog(@"%@%@", @"queryPart: ", queryPart);
    newUrl = [NSURL URLWithString:queryPart relativeToURL:baseURL];
  }
  if (@available(iOS 9.0, *)) {
    [_webView loadFileURL:newUrl allowingReadAccessToURL:[NSURL URLWithString:@"file:///"]];
  } else {
    return false;
  }
  return true;
}
```